bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c

```diff
@@ -507,53 +512,62 @@ esp_err_t bsp_display_new_with_handles(const bsp_display_config_t *config, bsp_l
-#elif CONFIG_BSP_LCD_HDMI_1280x720_60HZ
-    ESP_LOGI(TAG, "HDMI configuration for 1280x720@60HZ");
-    esp_lcd_dpi_panel_config_t dpi_config = LT8912B_1280x720_PANEL_60HZ_DPI_CONFIG();
-#elif CONFIG_BSP_LCD_HDMI_1280x800_60HZ
-    ESP_LOGI(TAG, "HDMI configuration for 1280x800@60HZ");
-    esp_lcd_dpi_panel_config_t dpi_config = LT8912B_1280x800_PANEL_60HZ_DPI_CONFIG();
-#elif CONFIG_BSP_LCD_HDMI_1920x1080_30HZ
-    ESP_LOGI(TAG, "HDMI configuration for 1920x1080@30HZ");
-    esp_lcd_dpi_panel_config_t dpi_config = LT8912B_1920x1080_PANEL_30HZ_DPI_CONFIG();
-#elif CONFIG_BSP_LCD_HDMI_1920x1080_60HZ
-    ESP_LOGI(TAG, "HDMI configuration for 1920x1080@60HZ");
-    /* This setting is not working yet, it is only for developing and testing */
-    esp_lcd_dpi_panel_config_t dpi_config = LT8912B_1920x1080_PANEL_60HZ_DPI_CONFIG();
-#else
-#error Unsupported display type
-#endif
-    dpi_config.num_fbs = CONFIG_BSP_LCD_DPI_BUFFER_NUMS;
+    esp_lcd_dpi_panel_config_t dpi_config;
+    const esp_lcd_dpi_panel_config_t dpi_configs[] = {
+        LT8912B_800x600_PANEL_60HZ_DPI_CONFIG(),
+        LT8912B_1024x768_PANEL_60HZ_DPI_CONFIG(),
+        LT8912B_1280x720_PANEL_60HZ_DPI_CONFIG(),
+        LT8912B_1280x800_PANEL_60HZ_DPI_CONFIG(),
+        LT8912B_1920x1080_PANEL_30HZ_DPI_CONFIG()
+    };
+
+    const esp_lcd_panel_lt8912b_video_timing_t video_timings[] = {
+        ESP_LCD_LT8912B_VIDEO_TIMING_800x600_60Hz(),
+        ESP_LCD_LT8912B_VIDEO_TIMING_1024x768_60Hz(),
+        ESP_LCD_LT8912B_VIDEO_TIMING_1280x720_60Hz(),
+        ESP_LCD_LT8912B_VIDEO_TIMING_1280x800_60Hz(),
+        ESP_LCD_LT8912B_VIDEO_TIMING_1920x1080_30Hz()
+    };
     lt8912b_vendor_config_t vendor_config = {
-#if CONFIG_BSP_LCD_HDMI_800x600_60HZ
-        .video_timing = ESP_LCD_LT8912B_VIDEO_TIMING_800x600_60Hz(),
-#elif CONFIG_BSP_LCD_HDMI_1024x768_60HZ
-        .video_timing = ESP_LCD_LT8912B_VIDEO_TIMING_1024x768_60Hz(),
-#elif CONFIG_BSP_LCD_HDMI_1280x720_60HZ
-        .video_timing = ESP_LCD_LT8912B_VIDEO_TIMING_1280x720_60Hz(),
-#elif CONFIG_BSP_LCD_HDMI_1280x800_60HZ
-        .video_timing = ESP_LCD_LT8912B_VIDEO_TIMING_1280x800_60Hz(),
-#elif CONFIG_BSP_LCD_HDMI_1920x1080_30HZ
-        .video_timing = ESP_LCD_LT8912B_VIDEO_TIMING_1920x1080_30Hz(),
-#elif CONFIG_BSP_LCD_HDMI_1920x1080_60HZ
-        /* This setting is not working yet, it is only for developing and testing */
-        .video_timing = ESP_LCD_LT8912B_VIDEO_TIMING_1920x1080_60Hz(),
-#else
-#error Unsupported display type
-#endif
         .mipi_config = {
             .dsi_bus = mipi_dsi_bus,
             .dpi_config = &dpi_config,
             .lane_num = BSP_LCD_MIPI_DSI_LANE_NUM,
         },
     };
+
+    /* DPI config */
+    switch (config->hdmi_resolution) {
+    case BSP_HDMI_RES_800x600:
+        ESP_LOGI(TAG, "HDMI configuration for 800x600@60HZ");
+        memcpy(&dpi_config, &dpi_configs[0], sizeof(esp_lcd_dpi_panel_config_t));
```

Comment:
since `vendor_config.mipi_config.dpi_config` is a `const esp_lcd_dpi_panel_config_t *`

Could we do

``` c
vendor_config.mipi_config.dpi_config = &dpi_configs[0];
```
without the memcpy?